### ISSUE TITLE
Example of training on tacotron outputs?

### ISSUE THREAD

**Issue by hadaev8** [kan-bayashi/ParallelWaveGAN#86]

**Comment by kan-bayashi** [comment]
Sorry, I have not yet parepare the example.
I will consider to implement it.

**Comment by hadaev8** [comment]
@kan-bayashi 
If I have dir with wave and spec npy file, can I just run parallel_wavegan/bin/train.py
Or I missed something?

**Comment by kan-bayashi** [comment]
If the length is adjusted, it may work correctly.

**Comment by hadaev8** [comment]
Just for sure, is it fine if my spectrogramms was made by bit different math? For example log scale instead of log10.

**Comment by kan-bayashi** [comment]
If you train the model from scratch, that is fine :)

**Comment by hadaev8** [comment]
I used published long checkpoint for initial weights.
And 400k steps parallel wavegan config.
In general, it is fine, but I hear in my headphones noise around "s" char.
Example: https://drive.google.com/open?id=1zu_J8ARwGTQcJSwA1kfG6QbrtAiYmWPm

Should you advise is it because of mel range (0-8000), general mel generating. or I need to train more, or something else?
I use Nvidia's tacotron/mel generation.
https://github.com/NVIDIA/tacotron2/blob/master/stft.py#L42

**Comment by kan-bayashi** [comment]
If you have many utterances, it is better to train from scratch with GTA outputs. In our experiment, that is better than finetuning.

**Comment by hadaev8** [comment]
@kan-bayashi 
Should you also advice what changes i need to make to generate 44.1k audio?

**Comment by kan-bayashi** [comment]
@hadaev8 How was the above training? Is it going well? 
I'm glad if you share your experiment, which is one of my motivations to make it open-source.

I've never tried 44.1k. I think it will be working if we just change the sampling rate, but it is better to use a large receptive field (e.g. large kernel size or a large number of layers). Have you already tried?

**Comment by hadaev8** [comment]
@kan-bayashi 
Well no, it was not successful.
I need to fix source data and make the dataset again.
Since sources in 44.1k, I decided to try without resampling, bigger win size (to fit it into tacotron VRAM budged) and 128 mel banks instead of 80.
Not sure if it will work because with around the same size of mel vocoder needs to do x2 work.

**Comment by hadaev8** [comment]
@kan-bayashi 
Just started training.
Is it ok to have such gap?
Maybe I need bigger model?

![alt text](https://i.imgur.com/uPNwlw6.png)

**Comment by kan-bayashi** [comment]
Sorry for the late reply. I think the model size is not related.
The loss value itself is not directly correlated to the perceptual quality, so it is better to check the generated wav at first.

**Comment by hadaev8** [comment]
Dont looks like good
https://i.imgur.com/PlmiBcP.png

Maybe i need to edit some other params? 
This is my config
https://drive.google.com/open?id=1Tk1MIODhDEyha-AdBtiGY0fm3XCxSq9P

Bsically, i edited upsampling/downsampling

**Comment by kan-bayashi** [comment]
It seems the training is failed.
I think training data is something wrong.
Please check the following items:
- The length of generated mel spectrogram are exactly matched with waveform. For example, sometimes there is a case where waveform is cut due to the removing silence but Mel is not.
- Normalization of the features is correctly performed.

**Comment by hadaev8** [comment]
Should processed spectrogram looks like this?
https://i.imgur.com/PNIpTv9.png

for length matching, I used this code
This is correct, right?

```
import librosa
import numpy as np

def process(path):
    data, sr = librosa.core.load(path, sr=44100)
    mel = np.load(path.replace('.wav', '.npy'))[1:-1]
    audio = np.pad(data, (0, 2048), mode="edge")
    audio = audio[:mel.shape[1] * 2048]
    np.save(path.replace('.wav', '-wave'), audio)
    np.save(path.replace('.wav', '-feats'), mel)
```

**Comment by kan-bayashi** [comment]
I cannot understand what the figure represents.

From your setting file, the `hop_size=512`.
Therefore, `len(mel) * 512 == len(audio)`.
And I'm not sure why you cut the first frame of the mel.
Please see the following part as the reference.
https://github.com/kan-bayashi/ParallelWaveGAN/blob/5486c15880dcef8d1b5188ffabb7c4cfe638350d/parallel_wavegan/bin/preprocess.py#L158-L171

If GTA outputs are exactly corresponding to the mel in the above example, that is OK.

**Comment by hadaev8** [comment]
@kan-bayashi 
This is spectrogram image.

Yes, I confused fft and win.
Now I started train with preprocessing from this repo.
Maybe its is my mitstake with win size or bad feature range in spectrograms, but now train went better than published logs.

https://i.imgur.com/7ZJ5dy2.png

**Comment by hadaev8** [comment]
Does step size and gamma params affect anything?

**Comment by kan-bayashi** [comment]
STFT-loss value itself is not directly proportional to the perceptual quality (See discussion #61). Please check the generated speech quality carefully.

**Comment by hadaev8** [comment]
How do you think, should I increase batch_max_steps for 44k sampling rate?
Or it is fine as it is?
